File: Benchmarks/enterprise/python/fastapi/testcode/benchmark_test_16540.py
# SPDX-License-Identifier: Apache-2.0
from fastapi import Request
from starlette.responses import HTMLResponse


async def BenchmarkTest16540(request: Request):
    upload_name = (await request.form()).get('upload', '')
    data = (lambda v: v.strip())(upload_name)
    return HTMLResponse('<html><body><h1>' + str(data) + '</h1></body></html>')

File: Benchmarks/normal/java/spring/testcode/BenchmarkTest16540.java
// SPDX-License-Identifier: Apache-2.0
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import org.springframework.web.bind.annotation.*;

@RestController
public class BenchmarkTest16540 {

    @PostMapping(path="/BenchmarkTest16540", consumes="multipart/form-data")
    public void BenchmarkTest16540(@RequestPart("multipart_field") String multipartField, HttpServletRequest request, HttpServletResponse response) throws Exception {
        String multipartValue = multipartField != null ? multipartField : "";
        try {
            Integer.parseInt(multipartValue);
        } catch (Exception ignored) {
        }
        response.getWriter().print("{\"status\":\"ok\"}");
    }
}
